cmd/aws-s3-provisioner.go

```diff
@@ -20,95 +20,108 @@ package main
 import (
 	"errors"
 	"flag"
+	"fmt"
 	"os"
 	"path"
 	"syscall"
 
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+
 	"github.com/golang/glog"
-	"github.com/yard-turkey/lib-bucket-provisioner/controller"
+	v1alpha1 "github.com/yard-turkey/lib-bucket-provisioner/pkg/apis/objectbucket.io/v1alpha1"
+	provisioner "github.com/yard-turkey/lib-bucket-provisioner/pkg/api/provisioner"
 
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
+
 )
 
 const (
-	provisionerName = "example.com/hostpath"
+	provisionerName = "aws-s3.io/bucket"
+	awsKey = "AWS_ACCESS_KEY_ID"
+	awsSecret = "AWS_SECRET_ACCESS_KEY"
 )
 
-type hostPathProvisioner struct {
+type awss3Provisioner struct {
 	// The directory to create PV-backing directories in
-	pvDir string
-
-	// Identity of this hostPathProvisioner, set to node's name. Used to identify
-	// "this" provisioner's PVs.
-	identity string
+	endpointUrl string
+	bucketName  string
+	accessKey   string
+	secretKey   string
 }
 
-// NewHostPathProvisioner creates a new hostpath provisioner
-func NewHostPathProvisioner() controller.Provisioner {
-	nodeName := os.Getenv("NODE_NAME")
-	if nodeName == "" {
-		glog.Fatal("env variable NODE_NAME must be set so that this provisioner can identify itself")
-	}
-	return &hostPathProvisioner{
-		pvDir:    "/tmp/hostpath-provisioner",
-		identity: nodeName,
+// NewAwsS3Provisioner creates a new aws s3 provisioner
+func NewAwsS3Provisioner() provisioner.Provisioner {
+	bname := os.Getenv("BUCKET_NAME")
+	glog.Infof("% is our bucket", bname)
+	user := os.Getenv(awsKey)
+	glog.Infof("% is our user", user)
+	pword := os.Getenv(awsSecret)
+	glog.Infof("% is our pword", pword)
+
+
+	//how do I fill these in?
```

Comment:
the OBC's storage class has a `parameters` section of key->value pairs. The storage admin and the provisioner work out how to define the object-store from the storage class. Eg. you could have a service (that contains the object-store's endpoint) name in the SC, or the endpoint itself, etc.